packages/nextjs/src/server/utils.ts

```diff
@@ -143,14 +143,22 @@ export const injectSSRStateIntoObject = <O, T>(obj: O, authObject: T) => {
   return { ...obj, __clerk_ssr_state };
 };
 
-// TODO: Make this a generic and move to @clerk/shared
+// TODO: Use the same function defined in @clerk/shared once the package is tree shakeable
 export const handleIsSatelliteBooleanOrFn = (opts: WithAuthOptions, url: URL): boolean => {
   if (typeof opts.isSatellite === 'function') {
     return opts.isSatellite(url);
   }
   return opts.isSatellite || false;
 };
 
+// TODO: Use the same function defined in @clerk/shared once the package is tree shakeable
+export const handleDomainStringOrFn = (opts: WithAuthOptions, url: URL): string => {
+  if (typeof opts.domain === 'function') {
+    return opts.domain(url);
+  }
+  return opts.domain || '';
+};
+
```

Comment:
❓ what do you think about using a generic function for both domain and isSatellite ?
```
type Fn<T> = (v: any) => T;
export function handleValueOrFn<T>(value: T | Fn<T>, url: URL, defaultValue: T): T {
    if (typeof value === 'function') {
        return value(url);
    }
    return value || defaultValue;
}

var request = { url: '' };
var opts = { isSatellite: true };
handleValueOrFn<boolean>(opts.isSatellite, new URL(request.url), false);

var request = { url: '' };
handleValueOrFn<boolean>((url: URL) => !!url, new URL(request.url), false);


var request = { url: '' };
var opts2 = { domain: 'true' };
handleValueOrFn<string>(opts2.domain, new URL(request.url), '');

var request = { url: '' };
handleValueOrFn<string>((url: URL) => url.host, new URL(request.url), '');
```

Reply:
For sure we can 😀, this was my thought as well but I decided to delay it. Glad that you brought this up.